### ISSUE TITLE
Make the camera use a PoseInterface, rather than an OrientationInterface

### ISSUE THREAD

**Issue by rpavlik** [OSVR/OSVR-Unity#15]
Some devices have positional trackers. Those that don't, we can add "analysis" to transform them so they do.

**Comment by DuFF14** [comment]
This was done, but exposed a bug in the Nod Ring demo where, since positional tracking wasn't being used, the Y position of the camera was being reset to 0 (the player's feet instead of where it was manually placed, above the neck). Might be a separate issue: do not reset the position from where it has been placed manually if PoseInterface is used but positional tracking is not being used.

**Comment by DuFF14** [comment]
Actually, that fix is using an OrientationInterface rather than a PoseInterface. Developer needs to be aware of that without looking at the docs. This just needs some obvious prefabs and/or starter scenes for each setup.

**Comment by rpavlik** [comment]
I will actually posit that the solution is neither of those, and that it's a design solution instead:

- camera should listen for pose callbacks and orientation callbacks, setting as much as they receive
- core should not emit pose callback when only orientation data is available

Does that make sense?

**Comment by DuFF14** [comment]
Yes, that works.

**Comment by rpavlik** [comment]
Is this done, or are we waiting on the normalization of messages from path tree.

**Comment by DuFF14** [comment]
The initial issue is done -- the camera uses a PoseInterface. 
When you say the core should not emit pose callback, do you mean there should be a callback in PoseInterface that doesn't have a position parameter?
Currently, if only orientation data is available, the position parameter comes in as (0,0,0).  The camera then sets its local position to (0,0,0), when often times the local position was originally set to something like (0,1.75,0) to be at eye-level. Could fix this by making the incoming position an offset from the local origin:

transform.localPosition = origin + position;

**Comment by rpavlik** [comment]
Well, shortly, once there are some improvements in the core, if a device only reports orientation, it won't send a pose callback with all zeros, so you'd presumably have both an orientation interface and a pose interface (same path) there.

One question that keeps nagging me is the "rigging" - even just rotation reports should result in translation and rotation of the eyes, since the eyes are not at the "pivot point" of the head (base of neck for estimate). For example, if you look straight down, you should see "in front" of your body, not down your spinal column/neck. (This is something Yuval has a lot more experience with than me, given orientation trackers prevalence on HMDs - I'm used to full pose reports on trackers attached to glasses) I am not sure if we should be including that per-tracker, or in some larger skeleton system (reporting the HMD tracker as "HMD" and automatically applying the transform to turn it into head or eyes). And, also there is the issue of whether applying a transform that results in a translation component to an orientation interface turns it into a full pose interface.

**Comment by russell-taylor** [comment]
This is the "Library Xforms" section of my potential_models.jpg document on Basecamp.  There is a good description of how to set things up to support this in http://dl.acm.org/citation.cfm?id=147201 (Implementation of flying, scaling and grabbing in virtual worlds).  It shows how to support consistent body motion within a "room" and includes eyes, head, tracker sensor, tracker source, etc.  How to do this without requiring a full-blown scene graph is something I have thought some about and could have a discussion on at some point if you'd like.

**Comment by rpavlik** [comment]
Non-paywalled link to that paper is here: http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.134.9852

**Comment by DuFF14** [comment]
Closing, as the camera no longer uses PoseInterface nor OrientationInterface now that poses come from DisplayConfig.
